### PR TITLE
[Modal] Removed inconsistent padding of modal section

### LINKED ISSUE #2061: Update modal sections to have uniform top and bottom spacing
Noticed while looking through the Figma UI kit. Sections within modals have uneven top and bottom spacing. 

**Current:**

slack-imgs

**Expected:**

Uniform spacing (`2rem top, bottom, left and right`).

cc @sarahill

### PR DESCRIPTION
**Pull request by mitchmaps**
### Why are these changes introduced?

Fixes #2061 

### What issue is this PR addressing?

The [Modal](https://polaris.shopify.com/components/overlays/modal) component currently has an inconsistent top and bottom spacing in the _section_ subcomponent.

### What is this pull request doing?

This very simple change removes a repeated function call that made the top padding set to the _base_ Polaris token value rather than the correct _loose_ one. Now the top and bottom padding for the section is set to _loose_.

### Before
image
image

### After
image
image

### CODE REVIEW

**Comment by kaelig** [comment]
Can this also be fixed in polaris-rails, and perhaps Shopify/shopify for consistency?

**Comment by AdamWhitcroft** [comment]
Hey @dleroux, not super sure I follow what you mean, but from my POV the `After` screenshot from above here is what I'd like. The content in the modal sits within uniform padding from top/bottom/left/right of the section.

**Comment by dleroux** [comment]
Ok perfect. Just to clarify, the `Sections` are normally 16px top and bottom and 20px for the sides. For some unknown reason, the first section only has 20px at the top. This PR should have all sections including the first one to have 20px all around.

File: src/components/Modal/components/Section/Section.scss [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 .Section {
   flex: 0 0 auto;
-  padding: spacing() spacing(loose);
+  padding: spacing(loose);
 
   &:first-of-type {
```
Comment (dleroux) [inline src/components/Modal/components/Section/Section.scss]: 🔥

**Comment by dleroux** [comment]
Thanks for doing this @mitchmaps! 🐑 🇮🇹 🎉 

Following @kaelig comment is this something you could tackle in rails?

**Comment by mitchmaps** [comment]
No problem @dleroux ! And yes I'll hop into that on Monday!

**Comment by mitchmaps** [comment]
@dleroux looks like this change has already been implemented in polaris-rails upon inspection :)

**Pull request merged**